diff --git a/dist/esm/chainHead/chainHead.mjs b/dist/esm/chainHead/chainHead.mjs
index df3d49d4dbb5e022d10b78e19886379f3cb86534..492eaf08129fe34cae5554109cfccb99992f5982 100644
--- a/dist/esm/chainHead/chainHead.mjs
+++ b/dist/esm/chainHead/chainHead.mjs
@@ -272,10 +272,16 @@ const getChainHead$ = (chainHead) => {
     }),
     shareReplay(1)
   );
-  merge(runtime$, bestBlocks$).subscribe({
+  // XXX rogue
+  pinnedBlocks$.subscribe({
     error() {
+
     }
   });
+  //merge(runtime$, bestBlocks$).subscribe({
+  //  error() {
+  //  }
+  //});
   let unfollow = noop;
   let started = false;
   let nSubscribers = 0;
diff --git a/dist/esm/chainHead/streams/get-runtime-creator.mjs b/dist/esm/chainHead/streams/get-runtime-creator.mjs
index 958800129934f41797f586bdf650117c94284d41..609c57997f259e093512dd65daec7e4b9cd27a8e 100644
--- a/dist/esm/chainHead/streams/get-runtime-creator.mjs
+++ b/dist/esm/chainHead/streams/get-runtime-creator.mjs
@@ -98,10 +98,10 @@ const getRuntimeCreator = (call$) => {
       },
       usages
     };
-    runtimeContext$.subscribe({
-      error() {
-      }
-    });
+    //runtimeContext$.subscribe({
+    //  error() {
+    //  }
+    //});
     return result;
   };
 };
diff --git a/dist/esm/chainHead/streams/pinned-blocks.mjs b/dist/esm/chainHead/streams/pinned-blocks.mjs
index cbdb1e0234e64f89ff1f508af59199b474163051..0cbf38ff2a7d7f7fab3691a2e263235b8a00e843 100644
--- a/dist/esm/chainHead/streams/pinned-blocks.mjs
+++ b/dist/esm/chainHead/streams/pinned-blocks.mjs
@@ -13,7 +13,7 @@ const createRuntimeGetter = (pinned, startAt) => {
   };
 };
 const deleteBlock = (blocks, blockHash) => {
-  blocks.get(blocks.get(blockHash).parent)?.children.delete(blockHash);
+  //blocks.get(blocks.get(blockHash).parent)?.children.delete(blockHash);
   blocks.delete(blockHash);
 };
 const deleteBlocks = (blocks, toDelete) => {
@@ -45,7 +45,7 @@ const getPinnedBlocks$ = (follow$, call$, blockUsage$, onUnpin, deleteFromCache)
           }
           const [finalizedHash] = event.finalizedBlockHashes.slice(-1);
           acc.finalized = acc.best = finalizedHash;
-          const lastIdx = event.finalizedBlockHashes.length - 1;
+          // const lastIdx = event.finalizedBlockHashes.length - 1;
           event.finalizedBlockHashes.forEach((hash, i) => {
             if (acc.blocks.has(hash)) {
               acc.blocks.get(hash).recovering = false;
@@ -53,9 +53,9 @@ const getPinnedBlocks$ = (follow$, call$, blockUsage$, onUnpin, deleteFromCache)
               acc.blocks.set(hash, {
                 hash,
                 parent: i === 0 ? event.parentHash : event.finalizedBlockHashes[i - 1],
-                children: new Set(
-                  i === lastIdx ? [] : [event.finalizedBlockHashes[i + 1]]
-                ),
+                //children: new Set(
+                //  i === lastIdx ? [] : [event.finalizedBlockHashes[i + 1]]
+                //),
                 runtime: hash,
                 refCount: 0,
                 number: event.number + i,
@@ -82,12 +82,12 @@ const getPinnedBlocks$ = (follow$, call$, blockUsage$, onUnpin, deleteFromCache)
             acc.blocks.get(hash).recovering = false;
           } else {
             const parentNode = acc.blocks.get(parent);
-            parentNode.children.add(hash);
+            // parentNode.children.add(hash);
             const block = {
               hash,
               number: parentNode.number + 1,
               parent,
-              children: /* @__PURE__ */ new Set(),
+              // children: /* @__PURE__ */ new Set(),
               runtime: event.newRuntime ? hash : parentNode.runtime,
               refCount: 0,
               recovering: false
@@ -116,7 +116,8 @@ const getPinnedBlocks$ = (follow$, call$, blockUsage$, onUnpin, deleteFromCache)
         case "finalized": {
           acc.finalized = event.finalizedBlockHashes.slice(-1)[0];
           const { blocks } = acc;
-          if (blocks.get(acc.best).number < blocks.get(acc.finalized).number)
+          //if (blocks.get(acc.best).number < blocks.get(acc.finalized).number)
+          if (blocks.get(acc.best) === undefined || blocks.get(acc.best).number < blocks.get(acc.finalized).number)
             acc.best = acc.finalized;
           acc.finalizedRuntime = acc.runtimes[blocks.get(acc.finalized).runtime];
           const { prunedBlockHashes: prunned } = event;
