diff --git a/node_modules/@substrate/connect/dist/mjs/connector/smoldot-light.js b/node_modules/@substrate/connect/dist/mjs/connector/smoldot-light.js
index 5bd8ebb..e1e1fc9 100644
--- a/node_modules/@substrate/connect/dist/mjs/connector/smoldot-light.js
+++ b/node_modules/@substrate/connect/dist/mjs/connector/smoldot-light.js
@@ -40,8 +40,8 @@ const getClientAndIncRef = (config) => {
     }
     const clientOptions = {
         portToWorker,
-        forbidTcp: true,
-        forbidNonLocalWs: true,
+        forbidTcp: false,
+        forbidNonLocalWs: false,
         maxLogLevel: 9999999,
         cpuRateLimit: 0.5,
         logCallback: (level, target, message) => {
